.github/workflows/renovate.yml

```diff
@@ -27,8 +25,14 @@ jobs:
           app-id: ${{ vars.BOT_APP_ID }}
           private-key: ${{ secrets.BOT_PRIVATE_KEY }}
       - env:
-          RENOVATE_TOKEN: ${{ secrets.RENOVATE_TOKEN || steps.generate-token.outputs.token }}
+          RENOVATE_TOKEN: ${{ steps.generate-token.outputs.token || secrets.RENOVATE_TOKEN }}
+          RENOVATE_GIT_AUTHOR: ${{ vars.RENOVATE_GIT_AUTHOR }}
         run: |
+          if [ -n "${{ steps.generate-token.outputs.token }}" ]; then
```

Comment:
I have several concern about the logic here:

1. According to the doc, `username` is also **optional, autodetected if not supplied**, have you ever tried whether that works?
2. If the template users are using PAT and no variable `RNOVATE_GIT_AUTHOR` is provided, we should use the default value, but seems the default value is removed.
3. If the template users provide a specific value to variable `RENOVATE_GIT_AUTHOR`, we should respect it whenever GitHub App is used.

Reply:
It needs to futhter verify feasibility on it.

Reply:
I have verified the following scenes:
- only bot token(not username): it detects username and sets the correct bot account.
- bot token and git author: it still resolves to use bot account.
- bot token and not git author(use yaml default author): 

I don't known how it creates a ghost user now and feel confused.🤣

Reply:
Is there any corresponding job logs or created pull requests?

Reply:
from actions [#3](https://github.com/msclock/smooth7zip/actions/runs/8699051435), [commit](https://github.com/msclock/smooth7zip/commit/2ea9e0170ddfbba3c85742f11bce44c05b4af380) renovate job run manually log says [it](https://github.com/msclock/smooth7zip/actions/runs/8699051435/job/23857017278#step:5:66) use configured git author. But the created pr [#1](https://github.com/msclock/smooth7zip/pull/1) [#2](https://github.com/msclock/smooth7zip/pull/2) both are authored by my configured bot account.

Reply:
From my understanding, they are two different things. The author of the pull request author is only determined by the token used when creating the pull request. So it will either be a personal account when using PAT or bot account when using token generated by GitHub App. The renovate configuration only affects the git author of the commits, as you can see in the `commits` tab [1][2].

[1] https://github.com/msclock/smooth7zip/pull/1/commits
[2] https://github.com/msclock/smooth7zip/pull/2/commits

Reply:
@huxuan Ah! I missed that flaws after several weeks.🤣

Reply:
Seems we have consumed too much energy on this, if necessary, we can have a online discussion about this.

Reply:
@huxuan  it's all right. just need to do trade-off. I have made another pr #470 and switch to just warn users. Feel free to alter it if you need any improvement.